torch_tvm/register.cpp

```diff
@@ -22,55 +23,93 @@ static std::string device_type = "cpu";
 
   // Register the tvm::CompilationGroup operator
   auto options = OperatorOptions().aliasAnalysis(AliasAnalysisKind::PURE);
-  RegisterOperators op({Operator(
-      tvm_sym,
-      [](const Node* node) {
-        auto cc = std::make_shared<TVMCompiler>(
-            node, opt_level, strict, device_type, device, host);
-        return [cc](Stack& stack) {
-          RECORD_FUNCTION("TVM", std::vector<c10::IValue>());
-          cc->run(stack);
-          return 0;
-        };
-      },
-      options)});
+  RegisterOperators op(
+      {Operator(tvm_sym,
+                [](const Node *node) {
+                  auto cc = std::make_shared<TVMCompiler>(
+                      node, opt_level, strict, device_type, device, host);
+                  return [cc](Stack &stack) {
+                    RECORD_FUNCTION("TVM", std::vector<c10::IValue>());
+                    cc->run(stack);
+                    return 0;
+                  };
+                },
+                options)});
 
   // Register the pass that fuses parts of the graph into
   // a tvm::CompilationGroup
-  RegisterPass pass([tvm_sym](std::shared_ptr<Graph>& g) {
+  RegisterPass pass([tvm_sym](std::shared_ptr<Graph> &g) {
     if (fusion_enabled) {
       CustomFuseGraph(g, isSupported, tvm_sym);
     }
   });
 
   // python API to enable and disable tvm fusion
-  m.def(
-      "enable",
-      [](int opt_level_,
-         bool strict_,
-         std::string device_type_,
-         std::string device_,
-         std::string host_) {
-        fusion_enabled = true;
-        strict = strict_;
-        opt_level = opt_level_;
-        device_type = device_type_;
-        device = device_;
-        host = host_;
-      },
-      py::arg("opt_level") = 2,
-      py::arg("strict") = false,
-      py::arg("device_type") = "cpu",
-      py::arg("device") = "llvm -mcpu=core-avx2",
-      py::arg("host") = "llvm -mcpu=core-avx2");
+  m.def("enable",
+        [](int opt_level_, bool strict_, std::string device_type_,
+           std::string device_, std::string host_) {
+          fusion_enabled = true;
+          strict = strict_;
+          opt_level = opt_level_;
+          device_type = device_type_;
+          device = device_;
+          host = host_;
+        },
+        py::arg("opt_level") = 2, py::arg("strict") = false,
+        py::arg("device_type") = "cpu",
+        py::arg("device") = "llvm -mcpu=core-avx2",
+        py::arg("host") = "llvm -mcpu=core-avx2");
 
   m.def("disable", []() { fusion_enabled = false; });
+  m.def("register_operator", [tvm_sym](std::string torch_name,
+                                       std::string relay_name) {
+    RegisterTVMOperator reg_tvm_operator(
+        {{Symbol::fromQualString("tvm::" + torch_name),
+          [relay_name](Node *node, tvm::Array<tvm::relay::Expr> inputs) {
+            auto op = tvm::relay::Op::Get(relay_name);
+            AT_ASSERT(op);
+            return tvm::relay::CallNode::make(op, inputs, tvm::Attrs(), {});
```

Comment:
does this mean that we currently can only register simple operators that does not require custom Attrs?

Reply:
yep, unfortunately.  There isn't a great way to expose the relay Attrs interface to python (as far as I can tell)